File: mint723/week_3/158p.java
// https://school.programmers.co.kr/learn/courses/30/lessons/12973

import java.util.*;

class Solution
{
    public int solution(String s)
    {
        Stack<Character> stack = new Stack<>();
        for(char c : s.toCharArray()){
            if(stack.isEmpty() || stack.peek()!=c){
                stack.add(c);
            }else{
                stack.pop();
            }
        }
        return stack.isEmpty()?1:0;
    }
}

File: kiv9908/week_3/158p.py
# 짝지어 제거하기
# https://school.programmers.co.kr/learn/courses/30/lessons/12973
def solution(s):
    stack = []
    answer = 0
    
    for i in s:
        if not stack:
            stack.append(i)
        else:
            if i == stack[-1]:
                stack.pop()
            else:
                stack.append(i)
    if not stack:
        answer = 1
            
    return answer